### ISSUE TITLE
binding redirect generation should not skip facade assemblies

### ISSUE THREAD

**Issue by kagemartin** [NuGet/Home#7393]
Visual Studio 15 creates binding redirects for reference assemblies.

```
<dependentAssembly>
  <assemblyIdentity name="System.Runtime" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
  <bindingRedirect oldVersion="0.0.0.0-4.1.2.0" newVersion="4.1.2.0" />
</dependentAssembly>
```

This is invalid because it forces the runtime to load the reference assembly rather than the framework assembly. This results in the following error:

```
An unhandled exception was generated during the execution of the current web request. Information regarding the origin and location of the exception can be identified using the exception stack trace below.
--

[BadImageFormatException: Cannot load a reference assembly for execution.]  [BadImageFormatException: Could not load file or assembly 'System.Runtime' or one of its dependencies. Reference assemblies should not be loaded for execution.  They can only be loaded in the Reflection-only loader context. (Exception from HRESULT: 0x80131058)]    System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks) +0    System.Reflection.RuntimeAssembly.nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks) +36    System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, Evidence assemblySecurity, RuntimeAssembly reqAssembly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks) +152    System.Reflection.RuntimeAssembly.InternalLoad(String assemblyString, Evidence assemblySecurity, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean forIntrospection) +77    System.Reflection.RuntimeAssembly.InternalLoad(String assemblyString, Evidence assemblySecurity, StackCrawlMark& stackMark, Boolean forIntrospection) +21    System.Reflection.Assembly.Load(String assemblyString) +28    System.Web.Configuration.CompilationSection.LoadAssemblyHelper(String assemblyName, Boolean starDirective) +38  [ConfigurationErrorsException: Could not load file or assembly 'System.Runtime' or one of its dependencies. Reference assemblies should not be loaded for execution.  They can only be loaded in the Reflection-only loader context. (Exception from HRESULT: 0x80131058)]    System.Web.Configuration.CompilationSection.LoadAssemblyHelper(String assemblyName, Boolean starDirective) +726    System.Web.Configuration.CompilationSection.LoadAllAssembliesFromAppDomainBinDirectory() +196    System.Web.Configuration.CompilationSection.LoadAssembly(AssemblyInfo ai) +45    System.Web.Compilation.BuildManager.GetReferencedAssemblies(CompilationSection compConfig) +172    System.Web.Compilation.BuildManager.GetPreStartInitMethodsFromReferencedAssemblies() +91    System.Web.Compilation.BuildManager.CallPreStartInitMethods(String preStartInitListPath, Boolean& isRefAssemblyLoaded) +111    System.Web.Compilation.BuildManager.ExecutePreAppStart() +156    System.Web.Hosting.HostingEnvironment.Initialize(ApplicationManager appManager, IApplicationHost appHost, IConfigMapPathFactory configMapPathFactory, HostingEnvironmentParameters hostingParameters, PolicyLevel policyLevel, Exception appDomainCreationException) +695  [HttpException (0x80004005): Could not load file or assembly 'System.Runtime' or one of its dependencies. Reference assemblies should not be loaded for execution.  They can only be loaded in the Reflection-only loader context. (Exception from HRESULT: 0x80131058)]    System.Web.HttpRuntime.FirstRequestInit(HttpContext context) +659    System.Web.HttpRuntime.EnsureFirstRequestInit(HttpContext context) +89    System.Web.HttpRuntime.ProcessRequestNotificationPrivate(IIS7WorkerRequest wr, HttpContext context) +190
```

**Comment by rrelyea** [comment]
@jainaashish  - dupe of 4.9 fix, right?
Link to issue / commit?

**Comment by jainaashish** [comment]
dupe of# https://devdiv.visualstudio.com/DevDiv/_workitems/edit/666788

Fixed with https://github.com/NuGet/NuGet.Client/pull/2416

It's already available in 15.9 preview3.
